metals/src/main/scala/scala/meta/internal/metals/ScalaVersions.scala

```diff
@@ -60,17 +60,25 @@ object ScalaVersions {
         latest != scalaVersion && SemVer
           .isCompatibleVersion(latest, scalaVersion)
       )
-      .getOrElse(
-        isLatestScalaVersion
-          .forall(ver => SemVer.isCompatibleVersion(ver, scalaVersion))
-      )
+      .getOrElse {
+        val versions =
+          if (isScala3Version(scalaVersion))
+            isLatestScalaVersion.filter(isScala3Version)
+          else
+            isLatestScalaVersion.filter(!isScala3Version(_))
+        versions.forall(ver => SemVer.isCompatibleVersion(ver, scalaVersion))
+      }
   }
 
   def isCurrentScalaCompilerVersion(version: String): Boolean =
     ScalaVersions.dropVendorSuffix(
       version
     ) == mtags.BuildInfo.scalaCompilerVersion
 
-  def scalaBinaryVersionFromFullVersion(scalaVersion: String): String =
-    scalaVersion.split('.').take(2).mkString(".")
+  def scalaBinaryVersionFromFullVersion(scalaVersion: String): String = {
+    if (scalaVersion == "3.0.0-M1")
+      scalaVersion
```

Comment:
I wasn't expecting the binary version here to be `3.0.0-M1`. At the same time, I was also wasn't expected `org.scala-lang:scala3-library_3.0.0-M1:3.0.0-M1` but rather just `org.scala-lang:scala3-library:3.0.0-M1`. 🤔

Reply:
Yeah, I was surprised by it too, but it seems that's how it's done currently :thinking: